pkg/sdk/urls.go

```diff
@@ -0,0 +1,10 @@
+package sdk
+
+import "fmt"
+
+// HostToWebsocketURL converts the provided host into a websocket URL that can
+// be used to subscribe to onchain events and query the chain via a client
+// context or send transactions via a tx client context.
+func HostToWebsocketURL(host string) string {
```

Comment:
I believe, it would be more robust to pass `*url.URL` down to this function which will ensure using the `Host` component to build the websocket URL. Its name would probably change too.

WDYT?

Reply:
I would disagree for multiple reasons:

1. We have some usages where we use a string directly. For example: `sdk.HostToWebsocketURL(queryHost)`
2. I don't want introduce an added dependency in this little package.
3. What if the source of the URL varies (url.URL, string, etc...)